classes/class3/homework/routes/cats.js

```diff
@@ -0,0 +1,90 @@
+var path = require('path');
+var Cat = require(path.join(__dirname, '../models/cat'));
+
+var colors = ['yellow',
+              'green',
+              'red',
+              'black',
+              'blue',
+              'orange',
+              'tabby',
+              'NaN',
+              'brown',
+              'teal',
+              'lemmony yellow',
+              'grey',
+              'gray',
+              'maroon',
+              'macaroon'];
+
+var names = ['George',
+             'Malyssa',
+             'Frederick',
+             'Johan',
+             'Suzy',
+             'Mary',
+             'Skags',
+             'Robert',
+             'Tinkerbell',
+             'Mr. Whiskers',
+             'King Fluffybutt']
+
+var cats = {};
+
+function all(req, res) {
+  var colorFilter;
+  Cat.find({colors: colorFilter}, function(err, cats) {
+    res.render('cats', {cats: cats});
+  });
+}
+
+function create(req, res) {
+  var color = colors[Math.floor(Math.random() * colors.length)];
+  var name = names[Math.floor(Math.random() * names.length)];
+  var age = Math.floor(Math.random() * name.length);
```

Comment:
Interesting choice, choosing the age based on name length? Also, this is somehow producing ages of only 0 or 1.